src/ucs/memory/rcache_int.h

```diff
@@ -28,34 +29,46 @@ enum {
-                                            This is a separate lock because we
-                                            may want to invalidate regions
-                                            while the page table lock is held by
-                                            the calling context.
-                                            @note: This lock should always be
-                                            taken **after** 'pgt_lock'. */
-    ucs_mpool_t              mp;       /**< Memory pool to allocate entries for
-                                            inv_q and page table entries, since
-                                            we cannot use regular malloc().
-                                            The backing storage is original mmap()
-                                            which does not generate memory events */
-    ucs_queue_head_t         inv_q;    /**< Regions which were invalidated during
-                                            memory events */
-    ucs_list_link_t          gc_list;  /**< list for regions to destroy, regions
-                                            could not be destroyed from memhook */
-
-    char                     *name;    /**< Name of the cache, for debug purpose */
+    ucs_rcache_params_t      params;      /**< rcache parameters (immutable) */
+
+    pthread_rwlock_t         pgt_lock;    /**< Protects the page table and all
+                                               regions whose refcount is 0 */
+    ucs_pgtable_t            pgtable;     /**< page table to hold the regions */
+
+
+    ucs_spinlock_t           lock;        /**< Protects 'mp', 'inv_q' and 'gc_list'.
+                                               This is a separate lock because we
+                                               may want to invalidate regions
+                                               while the page table lock is held by
+                                               the calling context.
+                                               @note: This lock should always be
+                                               taken **after** 'pgt_lock'. */
+    ucs_mpool_t              mp;          /**< Memory pool to allocate entries for
+                                               inv_q and page table entries, since
+                                               we cannot use regular malloc().
+                                               The backing storage is original mmap()
+                                               which does not generate memory events */
+    ucs_queue_head_t         inv_q;       /**< Regions which were invalidated during
+                                               memory events */
+    ucs_list_link_t          gc_list;     /**< list for regions to destroy, regions
+                                               could not be destroyed from memhook */
+
+    char                     *name;       /**< Name of the cache, for debug purpose */
```

Comment:
may be move it into position below lru?

Reply:
The `name` member wasn't added by this commit (it appears so only because a previous merge commit with master), and the `lru` was added as the last struct member.
Moving `name` to after `lru` means changing the order pre-existing struct members. Are you sure we want to do that?

Reply:
yes, why nor... as I can see this is internal datatype and we may ignore backward compatibility
@yosefe any ideas?

Reply:
Moved.